pkg/certifier/certify/certify.go

```diff
@@ -0,0 +1,200 @@
+//
+// Copyright 2022 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package certify
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/guacsec/guac/pkg/assembler"
+	"github.com/guacsec/guac/pkg/assembler/graphdb"
+	"github.com/guacsec/guac/pkg/certifier"
+	"github.com/guacsec/guac/pkg/certifier/osv"
+	"github.com/guacsec/guac/pkg/handler/processor"
+	"github.com/guacsec/guac/pkg/logging"
+	"github.com/neo4j/neo4j-go-driver/v4/neo4j/dbtype"
+)
+
+const (
+	BufferChannelSize int = 1000
+)
+
+func init() {
+	_ = RegisterCertifier(osv.NewOSVCertificationParser, certifier.CertifierOSV)
+}
+
+var (
+	documentCertifier = map[certifier.CertfierType]func() certifier.Certifier{}
+)
+
+// RegisterCertifier registers the active certifier for to generate attestations
+func RegisterCertifier(c func() certifier.Certifier, certifierType certifier.CertfierType) error {
+	if _, ok := documentCertifier[certifierType]; ok {
+		return fmt.Errorf("the certifier is being overwritten: %s", certifierType)
+	}
+	documentCertifier[certifierType] = c
+
+	return nil
+}
+
+// Certify queries the graph DB to get the packages to scan. Utilizing the registered certifiers,
+// it scans and generate vulnerability attestation for each package. Aggregating the results to the
+// top/root level package
+func Certify(ctx context.Context, client graphdb.Client, emitter certifier.Emitter, handleErr certifier.ErrHandler) error {
```

Comment:
hmm why is there a need for the graph DB client here, i think that should be part of the emitter.

Reply:
It is used in lines 139 (`getCompoents`) and 173(`getCompHelper`) to issue Cipher queries directly before we get to a GraphQL interface

Reply:
gotcha- we just had a chat about this and were talking about abstracting that into an interface instead so that we can more easily customize the components that we get (e.g.so we can configure a list of PURLs) - so moving those 2 functions behind an interface instead.

Reply:
@lumjjb Updated to add in interface to abstract out getting the components.